src/shogun/io/TBOutputFormat.cpp

```diff
@@ -99,19 +99,19 @@ tensorflow::Event TBOutputFormat::convert_vector(
 	tensorflow::Event e;
 	std::time_t now_t = convert_to_millis(value.second);
 	e.set_wall_time(now_t);
-	e.set_step(value.first.get_step());
+	e.set_step(value.first->get<int64_t>("step"));
 
 	tensorflow::Summary* summary = e.mutable_summary();
 	auto summaryValue = summary->add_value();
-	summaryValue->set_tag(value.first.get_name());
+	summaryValue->set_tag(value.first->get<std::string>("name"));
```

Comment:
I was not able to get any `std::string` (or even `char*`) using `get`, therefore this line (and some others) causes an error. However, it should be pretty reasonable to do that or?

Reply:
The issue is caused by this definition of `put` here (https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/base/SGObject.h#L391). Basically, when we try to extract a `string`, the `put` method will check only the options added with `SG_ADD_OPTION` and it will not consider parameter which are of type `string`.

Reply:
Yes there was a discussion about this in the pr that added it. Solution: check the options first and if they don’t work just call the standard out method.
@gf712